pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/rawingress_reconciler.go

```diff
@@ -0,0 +1,231 @@
+	IngressConfig *v1beta1api.IngressConfig
+}
+
+func NewRawIngressReconciler(client client.Client,
+	scheme *runtime.Scheme,
+	ingressConfig *v1beta1api.IngressConfig,
+	isvc *v1beta1api.InferenceService) (*RawIngressReconciler, error) {
+	ingress, err := createRawIngress(client, scheme, isvc, ingressConfig)
+	if err != nil {
+		return nil, err
+	}
+	return &RawIngressReconciler{
+		client:        client,
+		scheme:        scheme,
+		Ingress:       ingress,
+		URL:           createRawURL(client, isvc, ingressConfig),
+		IngressConfig: ingressConfig,
+	}, nil
+}
+func createRawURL(client client.Client,
+	isvc *v1beta1api.InferenceService,
+	ingressConfig *v1beta1api.IngressConfig) *knapis.URL {
+	url := &knapis.URL{}
+	url.Scheme = "http"
+	url.Host = isvc.Name + "-" + isvc.Namespace + "." + ingressConfig.IngressDomain
+	return url
+}
+func generateRule(podSpec *corev1.PodSpec,
+	ingressHost string,
+	componentName string) netv1.IngressRule {
+	pathType := netv1.PathType(netv1.PathTypeImplementationSpecific)
+	rule := netv1.IngressRule{
+		Host: ingressHost,
+		IngressRuleValue: netv1.IngressRuleValue{
+			HTTP: &netv1.HTTPIngressRuleValue{
+				Paths: []netv1.HTTPIngressPath{
+					{
+						PathType: &pathType,
+						Backend: netv1.IngressBackend{
+							Service: &netv1.IngressServiceBackend{
+								Name: componentName,
+								Port: netv1.ServiceBackendPort{
+									Number: constants.CommonDefaultHttpPort,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	return rule
+}
+
+func generateMetadata(isvc *v1beta1api.InferenceService,
+	componentType constants.InferenceServiceComponent) metav1.ObjectMeta {
+	var name string
+	switch componentType {
+	case constants.Transformer:
+		name = constants.DefaultTransformerServiceName(isvc.Name)
+	case constants.Explainer:
+		name = constants.DefaultExplainerServiceName(isvc.Name)
+	case constants.Predictor:
+		name = constants.DefaultPredictorServiceName(isvc.Name)
+	}
+	//get annotations from isvc
+	annotations := utils.Filter(isvc.Annotations, func(key string) bool {
+		return !utils.Includes(constants.ServiceAnnotationDisallowedList, key)
+	})
+	objectMeta := metav1.ObjectMeta{
+		Name:      name,
+		Namespace: isvc.Namespace,
+		Labels: utils.Union(isvc.Labels, map[string]string{
+			constants.InferenceServicePodLabelKey: isvc.Name,
+			constants.KServiceComponentLabel:      string(componentType),
+		}),
+		Annotations: annotations,
+	}
+	return objectMeta
+}
+
+//GenerateIngressHost return the config domain in configmap.IngressDomain
+func GenerateIngressHost(ingressConfig *v1beta1api.IngressConfig,
+	isvc *v1beta1api.InferenceService,
+	componentType string,
+	topLevelFlag bool) string {
+	if ingressConfig.IngressDomain == "" {
+		ingressConfig.IngressDomain = "example.com"
+	}
+	metadata := generateMetadata(isvc, constants.InferenceServiceComponent(componentType))
+	if topLevelFlag {
+		return metadata.Name + "-" + metadata.Namespace + "." + ingressConfig.IngressDomain
```

Comment:
Do we need to generate the host at component level?

Reply:
I think we need this, because the transformer and predictor have same path.